src/game_interpreter.cpp

```diff
@@ -1331,6 +1331,25 @@ bool Game_Interpreter::CommandChangeEquipment(RPG::EventCommand const& com) { //
 		}
 	} else {
 		for (const auto &actor : GetActors(com.parameters[0], com.parameters[1])) {
+			if (actor->HasTwoWeapons() && slot == RPG::Item::Type_shield && item_id != 0) {
+				return true;
+			}
+
+			if (Main_Data::game_party->GetItemCount(item_id, false) == 0 && !actor->IsEquipped(item_id)) {
+				Main_Data::game_party->AddItem(item_id, 1);
+			}
+
+			if (actor->HasTwoWeapons() && slot == RPG::Item::Type_weapon) {
+				RPG::Item* new_equipment = ReaderUtil::GetElement(Data::items, item_id);
+				RPG::Item* equipment1 = ReaderUtil::GetElement(Data::items, actor->GetWeaponId());
+				RPG::Item* equipment2 = ReaderUtil::GetElement(Data::items, actor->GetShieldId());
+
+				if (equipment1 && !equipment2 && !equipment1->two_handed && !new_equipment->two_handed) {
+					actor->ChangeEquipment(slot + 1, item_id);
```

Comment:
Is there a way to do all this logic in actor->ChangeEquipment? In the previous code the only thing the interpreter did was delegating the work to the actor class.
Also these "return true" are wrong because you are in a for-loop, use "continue".

Reply:
Oops, the "return -> continue" is right, I'll correct it

If we do this in actor->ChangeEquipment, then it will affect too if we change weapons from the menu.
i.e. I'm in the Equipment menu, my character with two weapons have only the first one equipped, and I just want to change that one, that's it. If I put that inside ChangeEquipment, then instead of replacing the old weapon for the new one, it will just insert the new one in the second slot and let the first one as it is.
This is a feature only the Event Command does.